### PR TITLE
Add cpu, memory and swap fileds in performance test result

### LINKED ISSUE #292: SMP should measure CPU and memory usage of mesh resources
**Current Behavior**
Current SMP metrics are focused primarily on network performance: latency, RPS, etc.

**Desired Behavior**
CPU and memory performance should be measured in addition to network performance.

---

**Alternatives / Additional Context**
Network performance is a very important aspect of a service mesh, but CPU and memory usage are important as well. Users will be able to use this information when determining how large their cluster needs to be.

### PR DESCRIPTION
**Pull request by gyohuangxin**
Signed-off-by: Huang Xin <xin1.huang@intel.com>

**Description**

This PR fixes #292

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by navendu-pottekkat** [comment]
@jbyers19 Could you review this?

File: protos/service_mesh_performance.proto
Comment (jbyers19) [inline protos/service_mesh_performance.proto:98]: Should mem values be `double` as well?

**Comment by jbyers19** [comment]
Referencing @navendu-pottekkat's [comment](https://github.com/service-mesh-performance/service-mesh-performance/issues/292#issuecomment-1002522323):
> ...updating the spec and a pluggable interface where users can bring in their own metrics plugin where Meshery will collect metrics like CPU and Memory use from it and add it to the test results, will be good to implement.

To make sure I understand the context of this PR, these changes are simply adding the ability to display CPU, mem, and swap information in the performance results output? It is up to the user to implement a hook to populate those fields, whether that be via Prometheus or some other tool?

File: protos/service_mesh_performance.proto
Comment (gyohuangxin) [reply protos/service_mesh_performance.proto:98]: @jbyers19 Thanks for reviewing. The mem and swap are calculated in `MB`, it's enough to use `int32` in my opinion.

**Comment by gyohuangxin** [comment]
> Referencing @navendu-pottekkat's [comment](https://github.com/service-mesh-performance/service-mesh-performance/issues/292#issuecomment-1002522323):
> 
> > ...updating the spec and a pluggable interface where users can bring in their own metrics plugin where Meshery will collect metrics like CPU and Memory use from it and add it to the test results, will be good to implement.
> 
> To make sure I understand the context of this PR, these changes are simply adding the ability to display CPU, mem, and swap information in the performance results output? It is up to the user to implement a hook to populate those fields, whether that be via Prometheus or some other tool?

It's correct, SMP defines the standard fields and users should populate fields by themselves. Here is an implement of SMP in Meshery: https://github.com/meshery/meshery/blob/master/handlers/load_test_preferences_handler.go

File: protos/service_mesh_performance.proto
Comment (jbyers19) [reply protos/service_mesh_performance.proto:98]: Since these will be used for reporting performance numbers, I believe a higher level of precision would be better. However, I'm new to the project and not familiar with precedents for these types of design decisions so will defer to others on this.

**Review by jbyers19 — APPROVED** [review]

File: protos/service_mesh_performance.proto
Comment (gyohuangxin) [reply protos/service_mesh_performance.proto:98]: Your comments are very helpful and constructive. I set these values as `int32` based on: 

https://github.com/service-mesh-performance/service-mesh-performance/blob/2d986e847ab68a88f9748d0705bbc88aff26cc07/protos/service_mesh_performance.proto#L107

@navendu-pottekkat Do you have any comments on this? Thanks.

Comment (navendu-pottekkat) [reply protos/service_mesh_performance.proto:98]: There doesn't seem to be a reason why we should stick to using `int32`. As @jbyers19 mentioned having a floating point precision makes sense as we want to capture data with higher levels of precision.

@sunkuranganath Do you have thoughts on this?

Comment (gyohuangxin) [reply protos/service_mesh_performance.proto:98]: @navendu-pottekkat Make sense, I changed `int32` to `double`.

Comment (gyohuangxin) [reply protos/service_mesh_performance.proto:98]: @navendu-pottekkat Can you review it again? Thanks.

**Review by navendu-pottekkat — APPROVED** [review]
@service-mesh-performance/spec-maintainers A new release should be made after merging this PR.

Please address any concerns. I think we can merge this PR.

**Review by warunicorn19 — APPROVED** [review]
Thank you @gyohuangxin

**Pull request merged**
